### PR TITLE
Story/11788 - Add Package Type support

### PR DESCRIPTION
**Pull request by PeterC10**
Package Type now displayed on Packages.

Pallet and Stillage Package Types added.

API endpoint added to retrieve Package Types.

Parent package type can now be set via `update_picking`.

### CODE REVIEW

File: addons/udes_stock/models/stock_picking_type.py [hunk 1]
```diff
@@ -443,6 +457,8 @@ def _prepare_info(self):
             - u_reserve_pallet_per_picking: boolean
             - u_max_reservable_pallets: int
             - u_selection_method: string
+            - u_check_package_type: boolean
+            - u_default_package_type_id: id
```
Comment (squah-u) [inline addons/udes_stock/models/stock_picking_type.py:461]: nit: should this say `int` like `default_location_dest_id`? I prefer `id` as it's more informative but for consistency `int` wins

File: addons/udes_stock/views/stock_picking_type.xml [hunk 2]
```diff
@@ -60,6 +60,8 @@
                     <field name="u_reserve_pallet_per_picking" />
                     <field name="u_max_reservable_pallets" />
                     <field name="u_selection_method" />
+                    <field name="u_check_package_type" />
+                    <field name="u_default_package_type_id" options="{'no_create': True}"/>
```
Comment (squah-u) [inline addons/udes_stock/views/stock_picking_type.xml:64]: formatting nit: space before the `/>`

**Review by squah-u — APPROVED** [review]

File: addons/udes_stock/models/stock_picking_type.py (hunk 1 above)
Comment (PeterC10) [reply addons/udes_stock/models/stock_picking_type.py:461]: Yep, that wasn't intentional! Good spot. :)

Comment (PeterC10) [reply addons/udes_stock/models/stock_picking_type.py]: Fixed. :)

File: addons/udes_stock/views/stock_picking_type.xml (hunk 2 above)
Comment (PeterC10) [reply addons/udes_stock/views/stock_picking_type.xml]: Fixed. :)

**Review by apatrick88 — APPROVED** [review]

**Pull request merged**